### ISSUE TITLE
HC rewrites RTSS 0 FPS frame limit option even though feature is set to disabled in HC settings

### ISSUE THREAD

**Issue by eugenov** [Valkirie/HandheldCompanion#773]
GPD
Winmax 2 2022

0.18.0.1b

HC rewrites RTSS options such as:
- FPS cap -> 0
- OSD support -> Off

Before HC start
![image](https://github.com/Valkirie/HandheldCompanion/assets/61651148/9558392d-7e6d-40e5-b2ac-7f8b69f4d981)
10 sec after HC start
![image](https://github.com/Valkirie/HandheldCompanion/assets/61651148/0406b853-341d-4fa0-bd2d-cd29d5bfa71a)

**Comment by CasperH2O** [comment]
Yes, can confirm. I found this also to be the reason the OSD is not shown.

**Comment by CasperH2O** [comment]
@micdah would you have time to look into this one? It seems after merging your PR, now, after an install of HC, the default value for the On-Screen Display Support on the RTSS side defaults to false. When people now select minimal extended or full in HC, nothing is shown due to this.

**Comment by micdah** [comment]
@CasperH2O sorry for my late reply, quite busy weekend 😄 

Will try to have a look later today and see if I can replicate the issue, alternatively we could remove the code which toggles OSD in RTS when switching overlay mode in HC, which should revert that behaviour if there is some bug within this codepath.

Changing framerate limit is the same behaviour as before the change, I believe that to be "as intended" right, as HC uses RTS for it's framerate limiter, so when it's disabled in HC, it changes it to 0 i RTSS.

I remember this as the behaviour also from before my PR.

**Comment by micdah** [comment]
@eugenov out of curiosity, what are your Handheld Companion settings when this occurs?

I'm specifically interested in what are the settings for "Framerate Limit" and "Overlay display level"?

From a quick glance at the code, it could look like "Framerate limit" might be set to disabled in HC and "Overlay display level" might be set to "Disabled"

**Comment by micdah** [comment]
I see, but if HC overlay is set to `None` then it will not kill HWInfo but let RTSS run, but disable OSD (_in current form_).

But if you switch to `External` this will leave both RTSS and HWInfo running, with OSD set to enabled in RTSS - this was the exact motivtion for me to add this mode, so that I could setup my own custom overlay in RTSS and not have HC interfere by killing HWInfo specifically.

I guess the issue with framerate limit is kind of the same, but different, in that HC needs to have framerate modes "enabled" (set to whatever HC is set to), "disabled" (set to 0 to disable) and "custom" which is leave it alone in RTSS and let the user pick a value.

In the current from HC has strong dependencies to RTSS/HWInfo and it's not widely supported that the user makes custom configurations in them (correct me if I'm wrong @CasperH2O). Like you shouldn't disable any sensors in HWInfo either as HC relies on some of them for it's internal logic

**Comment by CasperH2O** [comment]
@micdah thanks for investigating. @Valkirie found the time and made a fix for the OSD not being enabled, will release shortly.

**Comment by CasperH2O** [comment]
Release 0.18.0.2 fixes the OSD setting. As for the FPS limiter being set to 0, this is currently as expected but could use an improvement like how it was done for the TDP reverting to max and HC leaving it alone.

**Comment by anzenchitai** [comment]
Also, updating HC will cause it to stop auto launching RTSS and HWInfo.

**Comment by CasperH2O** [comment]
Code here requires a look at. https://github.com/Valkirie/HandheldCompanion/blob/main/HandheldCompanion/Platforms/RTSS.cs#L129

**Comment by CasperH2O** [comment]
Issue is back due to recent code changes.

**Comment by CasperH2O** [comment]
Fixed again for next EA release.

**Comment by MSeys** [comment]
@eugenov in what case does it not work?

If it's with Steam, it's an issue with RTSS as the FPS limit is seemingly applied to RTSS (so our part in HC is done correctly), but RTSS doesn't apply it correctly to the game if it initially came from Steam.

**Comment by CasperH2O** [comment]
Thanks @eugenov I'll double check this. To my knowledge AutoTDP does not set the FPS in RTSS, as that's what the actual framerate slider is for, but perhaps something else is going on.
